src/v/cluster/types.h

```diff
@@ -360,24 +362,31 @@ struct incremental_topic_updates {
     property_update<tristate<size_t>> retention_bytes;
     property_update<tristate<std::chrono::milliseconds>> retention_duration;
 
-    // Data-policy property is replicated by data_policy_frontend and handled by
-    // data_policy_manager.
-    property_update<std::optional<v8_engine::data_policy>> data_policy;
-
     friend bool operator==(
       const incremental_topic_updates&, const incremental_topic_updates&)
       = default;
 };
 
+// This class contains updates for topic properties which are replicates not by
+// topic_frontend
+struct incremental_topic_custom_updates {
+    // Data-policy property is replicated by data_policy_frontend and handled by
+    // data_policy_manager.
+    property_update<std::optional<v8_engine::data_policy>> data_policy;
+};
+
 /**
  * Struct representing single topic properties update
  */
 struct topic_properties_update {
+    // We need version to indetify request with custom_properties
+    static constexpr int32_t version = -1;
     explicit topic_properties_update(model::topic_namespace tp_ns)
       : tp_ns(std::move(tp_ns)) {}
 
     model::topic_namespace tp_ns;
     incremental_topic_updates properties;
+    incremental_topic_custom_updates custom_properties;
```

Comment:
I read this a couple times before understanding what's going on -- `properties` is the structure that will get serialized in the update_topic_properties_cmd, but `custom_properties` is not.

I'm not sure i can think of a much better name, but it would be useful to have a comment here explaining the distinction.